### DISCUSSION TITLE
Build completion with incomplete allocation

### DISCUSSION THREAD

**Discussion by HannesMorre** [inventree/InvenTree#6135] [Q&A] [answered]
When we order PCB assemblies at our EMS provider, most components are supplied by us from our stock. These are allocated to a build order in Inventree. Some components however are supplied by the EMS provider. These components are never registered in Inventree, thus they cannot be allocated to a build order.

Untill now BOM lists in Inventree did not include the components supplied by our EMS provider. 
I just found out that build orders can be completed with incomplete allocations, so this means that we can add the components supplied by our EMS provider to the BOM lists in Inventree.
The only problem there is that when I do this, the number calculated as "can build" will always be 0 since these components are never in stock. I did find out though that I can set them as "consumable" to fix this.

Is this the right way to go or is there another way to solve this problem?

**Comment by SchrodingersGat** [comment, marked as answer]
If you do not have stock "available" for the external parts then this is a sufficient way to achieve it. Another way would be to create external stock items which are located at your supplier, and use those.

**Reply by HannesMorre** [reply]
Your last suggestion seems a bit tricky for us, as we also have stock that belongs to us stored at that external location. These would easily get mixed up. I will use the "consumable" check for this. Thanks for your feedback!
